### DISCUSSION TITLE
Possible to search for a specific campsite within a campground?

### DISCUSSION THREAD

**Discussion by javagenki** [juftin/camply#51] [Q&A] [answered]
My family and I have a few favorite campsites at various campgrounds that are very hard to get. Is it possible to search and get notified on specific campsites within a campground or has someone found a good way to filter the notifications to search for specifics?

**Comment by juftin** [comment]
This is a really neat idea @javagenki - I will take a look into implementing this. It shouldn't be an insane amount of effort.

**Comment by juftin** [comment, marked as answer]
This was a great idea @javagenki, I just released `camply` `0.2.0` and the ability to search for a specific campsite has arrived along with some other neat stuff. Below is the documentation snippet from the new [ReadTheDocs](https://camply.readthedocs.io/en/latest/usage.html#searching-for-a-specific-campsite-by-id). Please let me know how this works for you!

#### Searching for a Specific Campsite by ID

Sometimes you have a favorite campsite inside your favorite campground. To search for just a specific campsite (and not just all campsites within a campground) you can give its ID to camply with the `--campsite` argument. For example, site `R035` in Many Glacier Campground, MT is close to a trailhead. Its URL is https://www.recreation.gov/camping/campsites/98363 , here we can see that it’s ID is 98363. You can search for one or many campsites by ID by supplying the `--campsite` argument. You can provide the `--campsite` argument once or multiple times to search for different campsites. Note, `--campsite` arguments override any `--rec-area` or `--campground` parameters provided.

```shell
camply campsites \
    --campsite 98363 \
    --start-date 2022-07-10 \
    --end-date 2022-07-18
```

**Reply by javagenki** [reply]
@juftin you rock! Going to check this out now! Thank you!

**Reply by javagenki** [reply]
@juftin Thanks again for this speedy update. Not to 'look a gift horse in the mouth' but the only issue I am finding now is how to setup my YAML files with campsite specific searches. I tried adding the 'campsites' variable to the yaml_utils.py file but I ended up causing errors - my Python skills are woefully lacking. Can we have a list of 'campsites' option similar to the 'campgrounds' list in the YAML file?

**Reply by juftin** [reply]
🤦‍♂️ yes of course @javagenki . Just released the [bugfix](https://github.com/juftin/camply/compare/v0.2.1...v0.2.2) tp PyPi and Docker: https://camply.readthedocs.io/en/latest/usage.html#using-a-yml-configuration-file-to-search-for-campsites

**Reply by javagenki** [reply]
Oh, I almost had it. Missed that second change in the file. Thank you!
